PetAdoptions/cdk/pet_stack/resources/use_cases/observability-getting-started.yml

```diff
@@ -0,0 +1,325 @@
+      Tags:
+        - Key: Name
+          Value: ObservabilityGettingStartedImmersionDay
+      VpcId: !Ref VPC
+  #-------------------------------------------------
+  # IAM ROLE FOR EC2 Instance
+  #-------------------------------------------------
+  InstanceRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: SSMCloudWatchInstanceRole
+      AssumeRolePolicyDocument:
+        Version: '2012-10-17'
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+                - ec2.amazonaws.com
+            Action:
+              - sts:AssumeRole
+      Path: '/'
+      ManagedPolicyArns:
+        - arn:aws:iam::aws:policy/CloudWatchAgentServerPolicy
+        - arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore
+        - arn:aws:iam::aws:policy/AmazonSSMPatchAssociation
+      Tags:
+        - Key: Name
+          Value: ObservabilityGettingStartedImmersionDay
+
+  InstanceProfile:
+    Type: AWS::IAM::InstanceProfile
+    Properties:
+      InstanceProfileName: SSMCloudWatchInstanceRole
+      Path: '/'
+      Roles:
+        - !Ref InstanceRole
+  #-------------------------------------------------
+  # EC2 instance using the latest Amazon Linux AMI
+  #-------------------------------------------------
+  LinuxEc2Instance1:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: t2.small
+      ImageId: !Ref LatestAmazonLinuxAmiId
+      NetworkInterfaces:
+        - AssociatePublicIpAddress: true
+          DeviceIndex: 0
+          GroupSet:
+            - !Ref InstanceSecurityGroup
+          SubnetId: !Ref SubnetPublic
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash
+            #Cloudformation Stack: ${AWS::StackName}
+            yum update -y
+            yum install -y tomcat tomcat-webapps
+            systemctl start tomcat.service
+      IamInstanceProfile: !Ref InstanceProfile
+      Tags:
+        - Key: Name
+          Value: AppServer1
+
+  LinuxEc2Instance2:
+    Type: AWS::EC2::Instance
+    Properties:
+      InstanceType: t2.small
+      ImageId: !Ref LatestAmazonLinuxAmiId
+      NetworkInterfaces:
+        - AssociatePublicIpAddress: true
+          DeviceIndex: 0
+          GroupSet:
+            - !Ref InstanceSecurityGroup
+          SubnetId: !Ref SubnetPublic
+      UserData:
+        Fn::Base64:
+          !Sub |
+            #!/bin/bash
+            #Cloudformation Stack: ${AWS::StackName}
+            yum update -y
+            yum install -y tomcat tomcat-webapps
+            systemctl start tomcat.service
+      IamInstanceProfile: !Ref InstanceProfile
+      Tags:
+        - Key: Name
+          Value: AppServer2
+
+  #-------------------------------------------------
+  # Lambda log group to store the fake logs in (ensures it will be deleted on stack deletion)
+  # Note: can't change the path for the lambda to log to
+  #-------------------------------------------------
+  LambdaLogGroup:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The default action when replacing/removing a resource is to delete it. Set explicit values for UpdateReplacePolicy / DeletionPolicy on potentially stateful resource: Resources/LambdaLogGroup